src/components/ImageGeneration/GenerationForm/GenerationForm.tsx

```diff
@@ -566,7 +569,7 @@ const GenerationFormInner = ({ onSuccess }: { onSuccess?: () => void }) => {
             <InputSegmentedControl name="aspectRatio" data={getAspectRatioControls(baseModel)} />
           </Stack>
           <Group position="apart" my="xs">
-            <InputSwitch name="nsfw" label="Mature content" labelPosition="left" />
+            <InputSwitch name="nsfw" label="Mature content" labelPosition="left" disabled={hasMinorResource} checked={hasMinorResource ? false : undefined}/>
```

Comment:
Just to be extra safe, we could and should do a check for this on the backend as well. In case there's some weirdo trying to make direct calls to the API

Reply:
i think this was enforced in the backed in an earlier branch:
https://github.com/civitai/civitai/blob/4cd151ccb5a98d3d39ad6f9c00b0d105faae00de/src/server/services/generation/generation.service.ts#L698-L701
this de wey?

Reply:
Yep, this is the way. Great that it was already implemented 👍